### PR TITLE
server: change db-table/{table_partition_id} http api (#10026)

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
cherry-pick PR #10026

### CODE REVIEW

**Comment by crazycs520** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

**Comment by crazycs520** [comment]
/run-all-tests tidb-test=release-2.1 tidb-private-test=release-2.1 tikv=release-2.1

File: server/http_handler_test.go [hunk 1]
```diff
@@ -584,6 +584,37 @@ func (ts *HTTPHandlerTestSuite) TestGetSchema(c *C) {
...
 	c.Assert(dbtbl.SchemaVersion, Equals, domain.GetDomain(se.(sessionctx.Context)).InfoSchema().SchemaMetaVersion())
+
+	db, err := sql.Open("mysql", getDSN())
+	c.Assert(err, IsNil, Commentf("Error connecting"))
+	defer db.Close()
+	dbt := &DBTest{c, db}
+
+	dbt.mustExec("create database if not exists test;")
+	dbt.mustExec("use test;")
+	dbt.mustExec("set @@tidb_enable_table_partition = 1")
```
Comment (bb7133) [inline server/http_handler_test.go]: why we need to set `tidb_enable_table_partition`(compared with https://github.com/pingcap/tidb/pull/10026)?

Comment (crazycs520) [reply server/http_handler_test.go]: Because release2.1 haven't enable create partition table default, But master default is enable.

Comment (bb7133) [reply server/http_handler_test.go]: Got it, thanks

**Pull request merged**